### PR TITLE
A couple of fixes to ensure promises are chained properly

### PR DESCRIPTION
**Pull request by jayongg**
Also a fix to check for number of favorites > 0 instead of > 1. This looks like a bug.  (> 0 is needed to check for existence, not > 1)

### CODE REVIEW

File: helpers.js [hunk 1]
```diff
@@ -1176,7 +1176,7 @@ const kodiFindFavourite = (request, favouriteName) => {
     return Kodi.Favourites.GetFavourites({ // eslint-disable-line new-cap
         properties: ['window','path','windowparameter']
     }).then((query) => {
-        if (!(query && query.result && query.result.favourites && query.result.favourites.length > 1)) {
```
Comment (keydon) [inline helpers.js]: good catch

File: helpers.js [hunk 2]
```diff
@@ -1019,7 +1019,7 @@ exports.kodiTestConnection = (request, response) => {
     return request.kodi.GUI.ShowNotification(param) // eslint-disable-line new-cap
     .then((result) => {
         console.log(param.message, result);
-        response.send(param.message);
+        return response.send(param.message);
```
Comment (keydon) [inline helpers.js]: `response` belongs to nodes express. the `send()` method does not return a promise or anything else. therefor its too much here :d

File: helpers.js [hunk 3]
```diff
@@ -759,7 +759,7 @@ exports.kodiPlayAlbum = (request, response) => { // eslint-disable-line no-unuse
     let Kodi = request.kodi;
 
     console.log(`Album request received to play "${albumTitle}"`);
-    kodiFindAlbum(albumTitle, Kodi)
+    return kodiFindAlbum(albumTitle, Kodi)
```
Comment (keydon) [inline helpers.js]: yea, here it was missing

**Comment by keydon** [comment]
thank you!

**Comment by jayongg** [comment]
Thanks!

Sent from mobile device
________________________________

**Pull request merged**